doc/agentvm.md

```diff
@@ -0,0 +1,59 @@
+# Agent virtual machine
+The agent, based on Red Hat CoreOS (RHCOS), communicates with the Agent Service and reports its status.
+The agent virtual machine is initialized using ignition, which configures multiple containers that run as systemd services. Each of these services is dedicated to a specific function.
+
+## Systemd services
+Follows the list of systemd services that can be found on agent virtual machine. All of the services
+are defined as quadlets. Quadlet configuration can be found in the [ignition template file](../data/config.ign.template).
+
+### planner-setup
+Planner-setup service is responsible for inicializing the volume with data, that are shared between `planner-agent` and `planner-agent-collector`.
+
+### planner-agent
+Planner-agent is a service that reports the status to the Agent service. The URL of the Agent service is configured in `$HOME/vol/config.yaml` file, which is injected via ignition.
+
+Planner-agent contains web application that is exposed via port 3333. Once user access the web app and enter the credentials of the vCenter, `credentials.json` file is created in the shared volume, and `planner-agent-collector` can be spawned.
+
+### planner-agent-opa
+Planner-agent-opa is a service that re-uses forklift validation container. The forklift validation container is responsible for vCenter data validation. When `planner-agent-collector` fetch vCenter data it's validated against the OPA server and report is shared back to Agent Service.
+
+### planner-agent-collector
+Planner-agent-collector service waits until user enter vCenter credentials, once credentials are entered the vCenter data are collected. The data are stored in `$HOME/vol/data/inventory.json`. Once `invetory.json` is created `planner-agent` service send the data over to Agent service.
+
+### podman-auto-update
+Podman auto update is responsible for updating the image of containers in case there is a new release of the image. We use default `podman-auto-update.timer`, which executes `podman-auto-update` every 24hours.
+
+## Troubleshooting
```

Comment:
```suggestion
## Troubleshooting Agent VM services
```
This is really great information for new users/contributers
I just suggest to make it clear in the header that this is relevant for the Agent VM